src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ImageController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\Command\AddProductImageCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\Query\GetProductImages;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\QueryResult\ProductImage;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\QueryResult\ProductImages;
+use PrestaShop\PrestaShop\Core\Image\Uploader\ImageUploaderInterface;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class ImageController extends FrameworkBundleAdminController
+{
+    public function uploadAction(int $productId, Request $request): JsonResponse
+    {
+        //@todo: handle multiple files
+        $imageFile = $request->files->all()[0];
```

Comment:
@matks, any suggestion to handle multple files?
1. I see ability to loop through files and invoke AddProductImageCommand for each of them :/
2. Thought about bulk command, but idk how should I pass the values (which should be collection/array of DTO's) when we only supposed to use scalar values. And anyway it would reuse the same single action handler ?

HELP! :smile: